File: Python/Kalkulator.py
import math

class Calculator:
    def __init__(self):
        self.pamiec_historia=[]

    def dodawanie(self, liczba1, liczba2):
        liczba3 = liczba1 + liczba2
        print(f"wynik dodawania: {liczba1} + {liczba2} = {liczba3}")
        self.pamiec_historia.append(f"wynik dodawania: {liczba1} + {liczba2} = {liczba3}")

    def odejmowanie(self, liczba1, liczba2):
        liczba3 = liczba1 - liczba2
        print(f"wynik odejmowania: {liczba1} - {liczba2} = {liczba3}")
        self.pamiec_historia.append(f"wynik odejmowania: {liczba1} - {liczba2} = {liczba3}")

    def mnozenie(self, liczba1, liczba2):
        liczba3 = liczba1 * liczba2
        print(f"wynik mnożenia: {liczba1} * {liczba2} = {liczba3}")
        self.pamiec_historia.append(f"wynik mnożenia: {liczba1} * {liczba2} = {liczba3}")

    def dzielenie(self, liczba1, liczba2):
        if liczba1 or liczba2 == 0:
            print("niemorzna dzielic przez zero")
        else:
            liczba3 = liczba1 / liczba2
            print(f"wynik dzielenia: {liczba1} / {liczba2} = {liczba3}")
            self.pamiec_historia.append(f"wynik dzielenia: {liczba1} / {liczba2} = {liczba3}")


    def potegowanie(self, liczba1, liczba2):
        liczba3 = pow(liczba1, liczba2)
        print(f"wynik potegowania: {liczba1}  {liczba2} tych dwuch liczb to: {liczba3}")
        self.pamiec_historia.append(f"wynik potegowania: {liczba1}  {liczba2} tych dwuch liczb to: {liczba3}")

    def pierwiastkowanie(self, liczba1, liczba2):
        liczba3 = liczba1 ** liczba2
        print(f"wynik pierwiastkowania to: {liczba2}^{liczba1} = {liczba3}")
        self.pamiec_historia.append(f"wynik pierwiastkowania to: {liczba2}^{liczba1} = {liczba3}")

    def historia(self):
        if not self.pamiec_historia:
            print("Historia jest pusta!")
        else:
            print(":--------------------------------:")
            for dzialanie in self.pamiec_historia:
                print(dzialanie)
            print(":--------------------------------:")

    def wyjscie(self):
        print("program zostaje zamkniety")

    def pobierz_2_liczby(self):
        liczba1 = float(input("podaj 1 liczbe"))
        liczba2 = float(input("podaj 2 liczbe"))
        return liczba1, liczba2


kalk = Calculator()

while True:
    print(":--------------------------------:")
    print("wpisz:")
    print("     1 - dodawanie")
    print("     2 - odejmowanie")
    print("     3 - mnorzenie")
    print("     4 - dzielenie")
    print("     5 - potegowanie")
    print("     6 - pierwiastkowanie")
    print("     7 - Otworz Historie")
    print("     8 - wyjsc z programu")
    print(":--------------------------------:")

    wybur = input("Wybierz dzialanie: ")

    if wybur == "1":
        liczba1, liczba2 = kalk.pobierz_2_liczby()
        kalk.dodawanie(liczba1, liczba2)
    elif wybur == "2":
        liczba1, liczba2 = kalk.pobierz_2_liczby()
        kalk.odejmowanie(liczba1, liczba2)
    elif wybur == "3":
        liczba1, liczba2 = kalk.pobierz_2_liczby()
        kalk.mnozenie(liczba1, liczba2)
    elif wybur == "4":
        liczba1, liczba2 = kalk.pobierz_2_liczby()
        kalk.dzielenie(liczba1, liczba2)
    elif wybur == "5":
        liczba1, liczba2 = kalk.pobierz_2_liczby()
        kalk.potegowanie(liczba1, liczba2)
    elif wybur == "6":
        liczba1, liczba2 = kalk.pobierz_2_liczby()
        kalk.pierwiastkowanie(liczba1, liczba2)
    elif wybur == "7":
        kalk.historia()
    elif wybur == "8":
        kalk.wyjscie()
        break

File: Rust/kalkulator.rs
use std::io;

fn main() {
    let mut historia: Vec<String> = Vec::new();
    
    loop {
        let mut input1 = String::new();
        let mut input2 = String::new();
        let mut wybur = String::new();

        println!("wpisz pierwszy numer: ");
        io::stdin()
            .read_line(&mut input1)
            .expect("niema liczby");
        let liczba1: f32 = input1.trim().parse().expect("niedostalem numera");

        println!("wpisz drugi numer: ");
        io::stdin()
            .read_line(&mut input2)
            .expect("niema liczby");
        let liczba2: f32 = input2.trim().parse().expect("niedostalem numera");

        println!("wybierz Dzialanie z wybranych:");
        println!("1 - dodawanie");
        println!("2 - odejmowanie");
        println!("3 - mnorzenie");
        println!("4 - dzielenie");
        println!("5 - potegowanie");
        println!("6 - pierwiastek tylko jenda liczba sie liczy");
        println!("7 - historia");
        println!("8 - wyjscie");
        io::stdin()
            .read_line(&mut wybur)
            .expect("niema wybora");

        if wybur.trim() == "1" {
            let liczba3 = liczba1 + liczba2;
            println!("wynik dodawania to {} + {} = {}", liczba1, liczba2, liczba3);
            historia.push(format!("{} + {} = {}", liczba1, liczba2, liczba3));
        }
        else if wybur.trim() == "2" {
            let liczba3 = liczba1 - liczba2;
            println!("wynik odejmowania to {} - {} = {}", liczba1, liczba2, liczba3);
            historia.push(format!("{} - {} = {}", liczba1, liczba2, liczba3));
        }
        else if wybur.trim() == "3" {
            let liczba3 = liczba1 * liczba2;
            println!("wynik mnorzenia to {} * {} = {}", liczba1, liczba2, liczba3);
            historia.push(format!("{} * {} = {}", liczba1, liczba2, liczba3));
        }
        else if wybur.trim() == "4" {
            if liczba2 == 0.0 {
                println!("Niemorzna dzielic przez zero");
                historia.push(format!("Blad obliczania"));
            }
            else {
                let liczba3 = liczba1 / liczba2;
                println!("wynik dzielenia to {} / {} = {}", liczba1, liczba2, liczba3);
                historia.push(format!("{} / {} = {}", liczba1, liczba2, liczba3));
            }
        }
        else if wybur.trim() == "5" {
            let liczba3 = liczba1.powf(liczba2);
            println!("wynik potegi to {}", liczba3);
            historia.push(format!("{}^2 = {}", liczba1, liczba3));
        }
        else if wybur.trim() == "6" {
            let liczba3 = liczba1.sqrt();
            println!("wynik pierwiastka to {}", liczba3);
            historia.push(format!("√{} = {}", liczba1, liczba3));
        }
        else if wybur.trim() == "7" {
            println!("historia: ");
            if historia.is_empty() {
                println!("historia jest pusta skurwysynie")
            }
            else {
                for h in &historia {
                    println!("{}", h);
                }
            }
        }
        else if wybur.trim() == "8" {
            println!("Program zostal zakonczony");
            break;
        }
        let mut kontynuj = String::new();
        println!("Wpisz 1 jerzeli chcesz kontynuowac, lub cos innego jerzeli chcesz zakonczyczyc:");
        io::stdin()
            .read_line(&mut kontynuj)
            .expect("blad z kontynuj");

        if kontynuj.trim() != "1" {
            println!("koniec Programu");
            break;
        }
    }
}
